### PR TITLE
Added a safe wrapper for mruby.

### PR DESCRIPTION
**Pull request by dragostis**

### CODE REVIEW

**Comment by dragostis** [comment]
- [ ] MRuby -> MRubyStruct
- [ ] MRubyStruct private
- [ ] MRuby trait with all functions
- [ ] `impl MRuby for Rc<RefCell<MRubyStruct>>`
- [ ] `&mruby` -> `&self`
- [ ] MRuby public & remove functions from lib

File: src/mruby.rs [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+            let mruby = Rc::new(RefCell::new(
+                MRuby {
+                    mrb: mrb,
+                    ctx: mrbc_context_new(mrb),
+                    classes: Box::new(HashMap::new()),
+                    methods: Box::new(HashMap::new())
+                }
+            ));
+
+            mrb_ext_set_ud(mrb, mem::transmute::<&Rc<RefCell<MRuby>>, *const u8>(&mruby));
```
Comment (apasel422) [inline src/mruby.rs]: I think this is the source of rust-lang/rust#31790. `&mruby` is a pointer to an `Rc`, but that pointer will be invalid once this method returns.

Comment (dragostis) [inline src/mruby.rs]: @apasel422 Any idea how I can send the `Rc` pointer to C so I can get it back?

Comment (apasel422) [inline src/mruby.rs]: You could transmute the `Rc` itself, or pass a pointer to the contained `RefCell` instead.

Comment (apasel422) [inline src/mruby.rs]:
```rust
mrb_ext_set_ud(mrb, mem::transmute::<Rc<RefCell<MRuby>>, *const u8>(mruby.clone()));
```

**Comment by dragostis** [comment]
Seems to be good. Thanks a lot, @apasel422. :smile:

**Comment by dragostis** [comment]
- [ ] Add `Option<T>`
- [ ] Fix `Rc` bug
- [ ] Override `Object.dup` in `def_class`
- [ ] Investigate `Proc`s
- [ ] Add mruby `Drop` test
- [ ] Write integration tests

**Comment by dragostis** [comment]
`MRuby` & `Value` don't drop.

**Pull request merged**